entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImpl.java

```diff
@@ -37,10 +47,19 @@ public class EntityChangeEventGeneratorImpl implements EntityChangeEventGenerato
 
   private final EventProducer<EntityChangeEventKey, EntityChangeEventValue>
       entityChangeEventProducer;
-  private Clock clock;
+  private final Map<String, List<String>> entityToAttributeChangeSkipMap;
+  private final Clock clock;
 
   EntityChangeEventGeneratorImpl(Config appConfig, Clock clock) {
     Config config = appConfig.getConfig(EVENT_STORE);
+    this.entityToAttributeChangeSkipMap =
+        config.hasPath(SKIP_ATTRIBUTES_CONFIG_PATH)
+            ? config.getConfigList(SKIP_ATTRIBUTES_CONFIG_PATH).stream()
+                .collect(
+                    Collectors.toUnmodifiableMap(
+                        conf -> conf.getString(SCOPE_PATH),
+                        conf -> conf.getStringList(ATTRIBUTES_PATH)))
+            : new HashMap<>();
```

Comment:
we don't want the map to be mutable in this path and immutable in the other - should always be immutable.